File: 0347 top-k-frequent-elements/solution.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer,Integer> map = new HashMap<>();
        for(int i:nums) map.put(i,map.getOrDefault(i,0)+1);
        List<Integer>[] b = new List[nums.length+1];
        for(int key:map.keySet()) {
            int val=map.get(key);
            if(b[val]==null) b[val]=new ArrayList<>();
            b[val].add(key);
        }
        int[] res = new int[k];
        int idx=0;
        for(int i=b.length-1;i>=0 && idx<k;i--) {
            if(b[i]!=null) {
                for(int num:b[i]) {
                    res[idx++]=num;
                    if(idx==k) break;
                }
            } 
        }
        return res;
    }
}

File: 0347 top-k-frequent-elements/solution.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
       var map = new Dictionary<int, int>();
        foreach (var p in nums)
        {
            if (map.ContainsKey(p))
             map[p]++;
             else
              map[p] = 1;
        }
        var buckets = new List<int>[nums.Length + 1];
        foreach(var kv in map) {
           int freq = kv.Value;
            if (buckets[freq] == null)
                buckets[freq] = new List<int>();
            buckets[freq].Add(kv.Key);
        }
        var res = new List<int>();
        for (int i = buckets.Length - 1; i >= 0 && res.Count < k; i--) {
            if (buckets[i] != null) {
                res.AddRange(buckets[i]);
            }
        }
        return res.Take(k).ToArray();
    }
}
